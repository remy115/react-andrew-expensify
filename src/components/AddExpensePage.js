import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export const AddExpensePage = (props) => (
  <div>
    <h1>Add Expense</h1>
    <ExpenseForm
      onSubmit={(expense) => {
        props.startAddExpense(expense);
        props.history.push('/');
      }}
    />
  </div>
);

const mapDispatch2Props=(dispatch)=>{
  return {
    startAddExpense:(expense)=>dispatch(startAddExpense(expense))
  }
}

export default connect(undefined,mapDispatch2Props)(AddExpensePage);
