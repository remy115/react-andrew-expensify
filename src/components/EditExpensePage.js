import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export const EditExpensePage = (props) => {
  return (
    <div>
      <ExpenseForm
        expense={props.expense}
        onSubmit={(expense) => {
          props.startEditExpense(props.expense.id, expense)
            .then(()=>{
              props.history.push('/');
            });
        }}
      />
      <button onClick={() => {
        props.startRemoveExpense({ id: props.expense.id })
          .then(()=>{

            props.history.push('/');
          });
      }}>Remove</button>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatch2Props=(dispatch)=>{
  return {
    startEditExpense:(id,expense)=>dispatch(startEditExpense(id, expense)),
    startRemoveExpense:(objId)=>dispatch(startRemoveExpense(objId))
  }
}

export default connect(mapStateToProps,mapDispatch2Props)(EditExpensePage);
