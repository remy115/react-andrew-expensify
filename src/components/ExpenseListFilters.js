import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';


export class ExpenseListFilters extends React.Component {
  constructor(props) {
    super(props);
    this.state={
      calendarFocused: null

    }
    this.onDatesChange=this.onDatesChange.bind(this);
    this.onFocusChange=this.onFocusChange.bind(this);

    // console.log('props.filters',props.filters);
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }
  render() {
    return (
      <div>
        <input
          type="text"
          name="text"
          value={this.props.filters.text}
          onChange={(e) => {
            this.props.setTextFilter(e.target.value);
          }}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={(e) => {
            if (e.target.value === 'date') {
              this.props.sortByDate();
            } else if (e.target.value === 'amount') {
              this.props.sortByAmount();
            }
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId="start"
          endDate={this.props.filters.endDate}
          endDateId="end"
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatch2Props=(dispatch)=>({
  setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
  setEndDate:(endDate)=>dispatch(setEndDate(endDate)),
  setTextFilter:(text)=>dispatch(setTextFilter(text)),
  sortByDate:()=>dispatch(sortByDate()),
  sortByAmount:()=>dispatch(sortByAmount())
})

export default connect(mapStateToProps,mapDispatch2Props)(ExpenseListFilters);
