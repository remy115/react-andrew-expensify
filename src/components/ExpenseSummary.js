import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import filteredExpenses from '../selectors/expenses';
import getTotalAmount from '../selectors/expenses-total';

export const ExpenseSummaryComp=({expenseCount, expensesTotal})=>{
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    return (
        <div className="page-header">
        <div className="content-container">
            <h1 className="page-header__title">Viewing <span>{expenseCount}</span> <span>{expenseWord}</span> totalling <span>${expensesTotal}</span></h1>
            <div className="page-header__actions">
                <Link className="button" to="/create">Add Expense</Link>
            </div>
        </div>
        </div>
    );
}

const mapState2Props=(store)=>{
    const filters=store.filters;
    const filteredExpenses1=filteredExpenses(store.expenses,filters);
    return {
        expenseCount:filteredExpenses1.length,
        expensesTotal:getTotalAmount(filteredExpenses1)
    }
}


export default connect(mapState2Props)(ExpenseSummaryComp);