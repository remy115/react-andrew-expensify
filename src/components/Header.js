import React from 'react';
import { Link } from 'react-router-dom';
import {startLogout} from '../actions/auth';
import {connect} from 'react-redux';

export const Header = ({startLogout}) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard" >
          <h1>Expensify</h1>
        </Link>
        <Link to="/create" activeClassName="is-active">Create Expense</Link>
        <button type="button" onClick={startLogout}>Logout</button>
      </div>
    </div>
  </header>
);

const mapDispatch2Props=(dispatch)=>{
  return {
    startLogout:()=>dispatch(startLogout())
  }
}

export default connect(null,mapDispatch2Props)(Header);