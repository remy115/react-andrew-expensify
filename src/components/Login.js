import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export class Login extends React.Component {
    constructor(props) {
        super(props)
        
        this.state={
            login:'',
            pass:''
        }        
        this.onSubmit=this.onSubmit.bind(this);
    }

    onSubmit(e) {
        e.preventDefault();
        if(!this.state.login || !this.state.pass) {
            return alert('Fill login and password!');
        }
    }

    onChange(e) {
        const name=e.target.name;
        const value=e.target.value;

        this.setState(()=>({
            [name]:value
        }));
    }

    render() {
        return (
            <div>
                <h2>Login here!</h2>
                {
                 (<button type="button" onClick={this.props.startLogin}>Log In</button>)
                    ||
                (<form onSubmit={this.onSubmit}>
                    <div>
                        <label>Login</label>
                        <input type="text" name="login"  />
                    </div>
                    <div>
                        <label>Password</label>
                        <input type="text" name="pass"  />
                    </div>
                </form>)
                }
            </div>
        )
    }
}

const mapDispatch2Props=(dispatch)=>{
    return {
        startLogin:()=>dispatch(startLogin())
    }
}

export default connect(null,mapDispatch2Props)(Login);