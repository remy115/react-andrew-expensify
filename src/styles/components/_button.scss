.button {
    background:$blue;
    border:none;
    border-radius:5px;
    color:#fff;
    display:inline-block;
    font-size:$font-size-large;
    font-weight:300;
    line-height:1;
    padding:$s-size;
    text-decoration:none;
}

.button--link {
    background:transparent;
}

.button--secondary {
    background:#808080;
}