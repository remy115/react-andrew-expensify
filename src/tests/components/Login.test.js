import React from 'react';
import {Login} from '../../components/Login';
import {shallow} from 'enzyme';

it('should render Login component',()=>{
    const wrapper=shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
})

it('should call startLogin on button click',()=>{
    const startLogin=jest.fn();
    const wrapper=shallow(<Login startLogin={startLogin} />);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
})