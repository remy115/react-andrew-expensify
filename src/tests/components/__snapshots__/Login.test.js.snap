// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login component 1`] = `
<div>
  <h2>
    Login here!
  </h2>
  <button
    type="button"
  >
    Log In
  </button>
</div>
`;
